heartbeat/monitors/wrappers/wrappers.go

```diff
@@ -64,55 +63,136 @@ func WrapLightweight(js []jobs.Job, stdMonFields stdfields.StdMonitorFields, mst
 			addMonitorDuration,
 			logMonitorRun(nil),
 		),
-		func() jobs.JobWrapper {
-			return makeAddSummary()
-		},
-		func() jobs.JobWrapper {
-			return addMonitorState(stdMonFields, mst)
-		},
 	)
-
+	var swrapped []jobs.Job
+	for _, wj := range wrapped {
+		swrapped = append(swrapped, addSummarizer(stdMonFields, mst, maxAttempts)(wj))
```

Comment:
Might be a `nit:`, but `StatefulFactory` overlaps significantly in intention with `jobs.WrapSeparately`. It's possible to get the same output by slightly editing JobWrapperFactory interface(unused atm) without these additional types:
```
wrapped := jobs.WrapAllSeparately(
    jobs.WrapAll(...),
    func(j jobs.Job) jobs.JobWrapper {
	s := newSummarizer(j, stdMonFields, mst, maxAttempts)
	return wrapSummarizer(s)
    },
```
If possible, I'd like to avoid introducing a new category of wrapper mechanism, so we could pick either one

Reply:
This is addressed in the latest version of this